examples/hosts/app-integration/document-recovery/src/dataController.ts

```diff
@@ -0,0 +1,66 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { EventEmitter } from "events";
+import { IValueChanged } from "fluid-framework";
+
+/**
+ * IDataController describes the public API surface for our data data object.
+ */
+export interface IDataController extends EventEmitter {
+    /**
+     * Get the data
+     */
+    readonly value: number;
+
+    /**
+     * Update data. Will cause a "dataChanged" event to be emitted.
+     */
+    updateData: () => void;
+
+    /**
+     * The dataChanged event will fire whenever someone updates the data, either locally or remotely.
+     */
+    on(event: "dataChanged", listener: () => void): this;
+}
+
+export const counterValueKey = "counterValue";
+
+interface DataControllerProps {
+    get: (key: string) => any;
+    set: (key: string, value: any) => void;
+    on(event: "valueChanged", listener: (args: IValueChanged) => void): this;
+    off(event: "valueChanged", listener: (args: IValueChanged) => void): this;
+}
+
+/**
+ * The DataController is our data object that implements the IDataController interface.
+ */
+export class DataController extends EventEmitter implements IDataController {
```

Comment:
Why do we extend EventEmitter if we're implementing IDataController that extends EventEmitter?

Reply:
EventEmitter not needed here.